app/src/renderer/model_factory.rs

```diff
@@ -4,20 +4,26 @@ use alloc::vec::Vec;
 use ogc_rs::print;
 use wavefront::Obj;
 
-use crate::raw_data_store::ModelName;
+use crate::raw_data_store::AssetName;
 
 use super::indexed_model::IndexedModel;
+use super::texture::Texture;
+use super::textured_model::TexturedModel;
 
 /**
  * All models must be defined in this list, which is filled at compile time.
  */
-const MODEL_KEYS: [ModelName; 1] = [ModelName::Suzanne];
+const TEXTURED_MODEL_DATA: [(AssetName, AssetName); 3] = [
```

Comment:
👍  Very good approach.

In a later PR we might split up the AssetName enum into the different kinds of files so different kinds of textures can only be used for their intended purpose, but that is for later.

Reply:
Good idea.